living_documentation_generator/generator.py

```diff
@@ -9,32 +32,53 @@
 
 
 from living_documentation_generator.github_projects import GithubProjects
+from living_documentation_generator.model.github_project import GithubProject
 from living_documentation_generator.model.config_repository import ConfigRepository
 from living_documentation_generator.model.consolidated_issue import ConsolidatedIssue
 from living_documentation_generator.model.project_issue import ProjectIssue
 from living_documentation_generator.utils.decorators import safe_call_decorator
 from living_documentation_generator.utils.github_rate_limiter import GithubRateLimiter
 from living_documentation_generator.utils.utils import make_issue_key
-from living_documentation_generator.utils.constants import (ISSUES_PER_PAGE_LIMIT, ISSUE_STATE_ALL,
-                                                            LINKED_TO_PROJECT_TRUE, LINKED_TO_PROJECT_FALSE)
+from living_documentation_generator.utils.constants import (
+    ISSUES_PER_PAGE_LIMIT,
+    ISSUE_STATE_ALL,
+    LINKED_TO_PROJECT_TRUE,
+    LINKED_TO_PROJECT_FALSE,
+)
 
 logger = logging.getLogger(__name__)
 
 
 class LivingDocumentationGenerator:
+    """
+    A class representing the Living Documentation Generator.
+
+    Attributes:
+        __github_instance (GitHub): The GitHub instance used for authentication and requests.
+        __github_projects_instance (GithubProjects): The GithubProjects instance used for project related operations.
+        __rate_limiter (GithubRateLimiter): The rate limiter used to control the rate of API calls.
+        __repositories (list[ConfigRepository]): List of config repositories to fetch from.
+        __project_state_mining_enabled (bool): Switch indicating if project state mining is enabled.
+        __output_path (str): The directory where the markdown pages will be stored.
+    """
 
     PROJECT_ROOT = os.path.dirname(os.path.abspath(__file__))
 
     ISSUE_PAGE_TEMPLATE_FILE = os.path.join(PROJECT_ROOT, "..", "templates", "issue_detail_page_template.md")
```

Comment:
considering using `os.pardir` instead of `".."`

Reply:
Great advice, change implemented in: [5db1778](https://github.com/AbsaOSS/living-doc-generator/pull/30/commits/5db1778f2ea5b37dd99c4b062d94c7c3ed9eded2)